web_ui/src/pages/annotator/components/explanation/use-inference-image.hook.test.tsx

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2022-2025 Intel Corporation
+// LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { renderHook } from '@testing-library/react';
+
+import { useLoadAIWebworker } from '../../../../hooks/use-load-ai-webworker/use-load-ai-webworker.hook';
+import { useInferenceImage } from './use-inference-image.hook';
+
+jest.mock('../../../../hooks/use-load-ai-webworker/use-load-ai-webworker.hook', () => ({
+    useLoadAIWebworker: jest.fn(),
+}));
+
+describe('useInferenceImage', () => {
+    const queryClient = new QueryClient();
+    const wrapper = ({ children }: { children: React.ReactNode }) => (
+        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+    );
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('returns resized image when worker is present', async () => {
+        const mockResize = jest.fn().mockResolvedValue('resized-image');
+        const mockInstance = { resize: mockResize };
+        const mockWorker = {
+            InferenceImage: jest.fn().mockResolvedValue(mockInstance),
+        };
+        jest.mocked(useLoadAIWebworker).mockReturnValue({ worker: mockWorker });
+
+        const { result } = renderHook(() => useInferenceImage(100, 200), { wrapper });
+
+        const imageData = { foo: 'bar' } as unknown as ImageData;
```

Comment:
We could use a "real" image data object here,

```suggestion
        const imageData = new ImageData(100, 100)
```